cmd/cluster/core/create.go

```diff
@@ -406,6 +410,12 @@ func prototypeResources(opts *CreateOptions) (*resources, error) {
 	}
 	prototype.Cluster.Spec.Networking.ServiceNetwork = serviceNetworkEntries
 
+	var machineNetworkEntries []hyperv1.MachineNetworkEntry
+	for _, cidr := range opts.MachineCIDR {
+		machineNetworkEntries = append(machineNetworkEntries, hyperv1.MachineNetworkEntry{CIDR: *ipnet.MustParseCIDR(cidr)})
```

Comment:
The idea behind the validate/complete pattern is that the types coming out the other end to be used in product code don't need to do things like `ipnet.MustParseCIDR` - parse the value in `Complete()` and store the parsed CIDR as `net.IPNet` directly, so the code here can be certain it has the correct inputs. This has the nice side-effect of not writing code that can panic. 

A nice read on the thought behind it: https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/

Reply:
Thanks for the link. We're now using the same pattern as ClusterCIDR and ServiceCIDR above.